uqcsbot/api.py

```diff
@@ -56,16 +63,27 @@ def __call__(self, **kwargs) -> dict:
 
         Attempts to retry the API call if rate-limited.
         """
-        fn = partial(self._client.api_call, self._method, **kwargs)
+        def get_fn(call_type):
+            client = getattr(self, f'_{call_type}_client')
+            method = getattr(client, self._method.replace('.', '_'))
+            return partial(method, **kwargs)
         retry_count = 0
+        call_type = _CLIENT_METHOD_REGISTRY.get(self._method, 'bot')
         while retry_count < 5:
-            result = fn()
+            try:
+                result = get_fn(call_type)()
```

Comment:
So we cleverly retrieve a partially applied function so it can immediately be called each time. Can we just call the method instead of partially applying it in `get_fn`?

Reply:
Good catch